### PR TITLE
feat: implement `fetchIdentityIdsByPublicKeys` method

### PR DESCRIPTION
**Pull request by jawid-h**
## Issue being fixed or feature implemented

As per new identity DIP we need a method to retrieve identity id by any public key hash

## What was done?

Added `DriveStateRepository#fetchIdentityIdsByPublicKeys` method

## Breaking Changes

None.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: lib/dpp/CachedStateRepositoryDecorator.js
Comment (shuplenkov) [inline lib/dpp/CachedStateRepositoryDecorator.js:66]: maybe identityPublicKeyHashes?

Comment (shumkov) [reply lib/dpp/CachedStateRepositoryDecorator.js:66]: Agree `fetchIdentityIdsByPublicKeyHashes`

File: lib/dpp/CachedStateRepositoryDecorator.js
Comment (shumkov) [inline lib/dpp/CachedStateRepositoryDecorator.js:55]: Do we still need it?

Comment (jawid-h) [reply lib/dpp/CachedStateRepositoryDecorator.js:55]: @shumkov I would like to leave it to keep tests working. After we merge it, we update tests, and then remove this method as a separate PR. This will save us some nerves.

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**
